### DISCUSSION TITLE
Use wlan0 and wlan1 rather than eth0 for my AP

### DISCUSSION THREAD

**Discussion by BSBoatia** [RaspAP/raspap-webgui#878] [Q&A] [answered]
Hello,

I'm at the end of my wits and hoping this community can help me. 

My objective: 
Provide access point and internet connectivity with two wireless interfaces (one as AP, one as client). The client might connect to various APs, as I am moving around with the Pi using whatever wifi is available.

Environment:
- Raspberry Pi 4 with Raspbian OS Lite (Buster - Kernel Version: 5.10.17-v7l+ #1403)
- External wlan adapter: Edimax 7811Un (V2 - 7392:b811 Edimax Technology Co., Ltd )

Note:
I'm aware of https://docs.raspap.com/faq/#interfaces (Can I use wlan0 and wlan1 rather than eth0 for my AP?)

Here's the thing:
On a clean Buster install
 - Install RaspAP, configure AP using wlan0 (internal pi adapter) **works fine**.
 - Edimax 7811Un not recognized as wlan interface - I'm guessing the FAQ refers to the 7392:**7**811 version, not 7392:**b**811). However, after buying 15+ wlan dongles, I gave up getting a 7392:**7**811 . They may simply not be available in Germany.
 - Install 7392:**b**811 (Realtek 8188eu) driver for kernel version 5.10.17-v7l+ #1403 (http://downloads.fars-robotics.net/wifi-drivers/8188eu-drivers/)

After reboot:
- AP still works fine using wlan0 (however, this is now the Edimax adapter)
- the internal pi adapter becomes wlan1 (I'm assuming they might switch back and forth, unless _"predictable network i/f names"_ is enabled via raspi-config. Does that matter?

- wlan1     IEEE 802.11  ESSID:off/any  
          Mode:Managed  Access Point: Not-Associated   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Power Management:on

The FAQ says, _"Select wlan1 as the AP interface in Hotspot and Save Settings."_. I cannot do that, as the RaspAP behavior is somewhat strange:
- I can select wlan1 as hotspot in basic settings -> interface. 
- However, I cannot configure a static IP in DHCP server settings for wlan1; RaspAP simply won't allow me to check "Static IP" it in the GUI. RapsAP is apparently dead set on keeping wlan0 to be the AP.

Anyhow, keeping wlan0 as AP and using wlan1 as client seems to work. I can scan for networks and connect successfully with wlan1.

So far, so good but I do not have internet connectivity via the pi's AP. I'm guessing it's about bullet 1. in the FAQ - _"Bind the two adapters with the interface names wlan0 and wlan1"_ 

How do I bind the two interfaces? Especially keeping in mind that the client might be assigned any IP/ subnet from the external wifi.

Your input is much appreciated!

Thanks & best,
Benny

**Comment by BSBoatia** [comment, marked as answer]
I have made a little bit of progress which I wanted to share. The issue seems to be related the the routes:

**Kernel IP routing table**
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0            192.168.8.1     0.0.0.0                UG    202      0        0 eth0
0.0.0.0            10.10.10.1        0.0.0.0                UG    303      0        0 wlan0
0.0.0.0            192.168.8.1     0.0.0.0                UG    304      0        0 wlan1
10.10.10.0       0.0.0.0            255.255.255.0   U      303       0        0 wlan0
192.168.8.0    0.0.0.0           255.255.255.0    U      202       0        0 eth0
192.168.8.0    0.0.0.0           255.255.255.0    U      304       0        0 wlan1

Deleting the 0.0.0.0 routes for eth0 and wlan0 solves the issue - at least until reboot.
```
sudo route del -net 0.0.0.0 gw 10.10.10.1 netmask 0.0.0.0 dev wlan0
sudo route del -net 0.0.0.0 gw 192.168.8.1 netmask 0.0.0.0 dev eth0
```

I have tried to use reduce interface priority via ifmetric. 
```
sudo ifmetric wlan0 2
sudo ifmetric eth0 1 
```
which does not seem to have any affect however.

Any ideas how to remove the 0.0.0.0 routes for eth0 and wlan0 in a way that sustains a reboot? Ideally, this change won't be overwritten by changes via RaspAP gui. Or is there a better way to achieve the desired result?

Thanks & best,
Benny

**Reply by BSBoatia** [reply]
In contrast to what http://0pointer.de/lennart/projects/ifmetric/ says, the interface with the lowest metric has highest priority. 

`sudo ifmetric wlan1 100` 

does the trick until reboot. Adding the following to /etc/dhcpcd.conf makes the change permanent 

```
interface wlan1
metric 100
```

Works for me. HAPPY!

**Reply by billz** [reply]
Glad to hear you found a solution.

> the interface with the lowest metric has highest priority

Correct.

> is there a better way to achieve the desired result?

The interface metric can be set in the **DHCP server settings** UI. Your manual changes won't be overwritten by RaspAP.
